biz.aQute.repository/src/aQute/bnd/repository/maven/pom/provider/BndPomRepository.java

```diff
@@ -75,21 +80,90 @@
 	private String				queryUrl;
 	private ScheduledFuture<?>	pomPoller;
 
-	public synchronized void init() {
+	private String				status;
+
+	private PromiseFactory		promiseFactory		= new PromiseFactory(Executors.newSingleThreadExecutor());
+
+	@SuppressWarnings("deprecation")
+	private boolean init() {
 		try {
-			if (inited)
-				return;
-			inited = true;
+			if (initialized == null) {
```

Comment:
Generally you make sure that this is initialized before you call the method. You can create the promise in prepare() since this is _always_ called before any of the methods that can call init().

Reply:
I already do all this in the prepare method, but a lot of the existing tests didn't work this way. So I decided to do the extra check, so it would still work as it did before. As this is public API, I did not want to make it too hard on other people using it. I reasoned, that this check will not be too expensive and should be tolerable.

Reply:
yeah, for tests you sometimes have to compromise, ok